rust/src/ftp/mod.rs

```diff
@@ -88,6 +124,40 @@ named!(pub ftp_epsv_response<u16>,
         )
 );
 
+// EPRT |2|2a01:e34:ee97:b130:8c3e:45ea:5ac6:e301|41813|
+named!(pub ftp_active_eprt<u16>,
+       do_parse!(
+            tag!("EPRT") >>
+            take_until_and_consume!("|") >>
+            take_until_and_consume!("|") >>
+            take_until_and_consume!("|") >>
+            port: getu16 >>
+            tag!("|") >> tag!("\0") >>
+            (
+                port
+            )
+        )
+);
+
+#[no_mangle]
+pub extern "C" fn rs_ftp_active_eprt(input: *const u8, len: u32) -> u16 {
+    let buf = unsafe{std::slice::from_raw_parts(input, len as usize)};
+    match ftp_active_eprt(buf) {
+        Ok((_, dport)) => {
+            return dport;
+        },
+        Err(nom::Err::Incomplete(_)) => {
+            let buf = unsafe{std::slice::from_raw_parts(input, len as usize)};
```

Comment:
I wonder how Rust deals with this in the non-debug case: will this lead to warnings of unused vars?

If so, maybe it should be:
```rust
let _buf = unsafe{std::slice::from_raw_parts(input, len as usize)};
SCLogDebug!("eprt incomplete: '{:?}'", String::from_utf8_lossy(_buf));
```

Reply:
Ah no, it should not be necessary.

Reply:
@victorjulien , @jasonish : I think it is possible to enable only a subset of clippy. This would allow a smooth transition (enabling more warnings when code is ready) and can be checked in travis CI.

I'll have a look

Reply:
@victorjulien @jasonish : a simple check shows that there is ~ 130 functions that should be marked unsafe. I can fix that, but would you agree with a fix-everything patch fixing all these functions and adding a travis check ?

Reply:
@chifflier In general there are some clippy warnings I don't agree with.  Right now I'm more concerned with our usage of unsafe.  Would it be a good idea to mark all these functions as unsafe, instead of just around the pointer dereference?  I think it make sense to. Thanks.

Reply:
I'm indifferent to a fix everything approach.  For was initially thinking we'd do it just for new functions, then updating existing ones as we go.  But that doesn't make it harder to look through the clippy output.

Reply:
Clippy is quite flexible, so it can be used to check only the warnings you agree with. This is what I'm proposing to add, if you want.

Specifically on the unsafe functions, I tend to agree with clippy: functions dereferencing unsafe (extern C) pointers should be 1) minimal 2) marked as unsafe function, so the caller knows it.